colorbleed/fusion/lib.py

```diff
@@ -38,3 +42,118 @@ def update_frame_range(start, end, comp=None, set_render_range=True):
+        if versions:
+            # ensure the "v" is not included and convert to ints
+            int_versions = [int(v[1:]) for v in versions]
+            version_int += max(int_versions)
+
+    return "v{:03d}".format(version_int)
+
+
+def update_savers(comp, session, project):
+    """Update all savers of the current comp to ensure the output is correct
+
+    Args:
+        comp (object): current comp instance
+        session (dict): the current Avalon session
+        project (dict): the project document from the database
+
+    Returns:
+         None
+    """
+
+    template = project["config"]["template"]["work"]
+    template_work = pipeline._format_work_template(template, session)
+
+    render_dir = os.path.join(os.path.normpath(template_work), "renders")
+    version_folder = get_next_version_folder(render_dir)
+    renders_version = os.path.join(render_dir, version_folder)
+
+    comp.Print("New renders to: %s\n" % render_dir)
+
+    with avalon.fusion.comp_lock_and_undo_chunk(comp):
+        savers = comp.GetToolList(False, "Saver").values()
+        for saver in savers:
+            filepath = saver.GetAttrs("TOOLST_Clip_Name")[1.0]
+            filename = os.path.basename(filepath)
+            new_path = os.path.join(renders_version, filename)
+            saver["Clip"] = new_path
+
+
+def switch(asset_name):
+    """Switch the current containers of the comp to the other asset (shot)
+
+    Args:
+        asset_name (str): name of the asset (shot)
+
+    Returns:
+        comp (PyObject): the comp instance
+
+    """
+
+    assert asset_name, "Function requires asset name"
+
+    host = api.registered_host()
+    assert host, "Host must be installed"
+
+    # Get current project
+    project = io.find_one({"type": "project",
+                           "name": api.Session["AVALON_PROJECT"]})
+
+    # Assert asset name exists
+    # It is better to do this here then to wait till switch_shot does it
+    asset = io.find_one({"type": "asset", "name": asset_name})
+    assert asset, "Could not find '%s' in the database" % asset_name
+
+    # Use the current open comp
+    current_comp = avalon.fusion.get_current_comp()
+    assert current_comp is not None, "Could not find current comp"
+
+    containers = list(host.ls())
+    assert containers, "Nothing to update"
+
+    representations = []
+    for container in containers:
+        try:
+            representation = colorbleed.switch_item(container,
+                                                    asset_name=asset_name)
+            representations.append(representation)
+            log.debug(str(representation["_id"]) + "\n")
+        except Exception as e:
+            log.debug("Error in switching! %s\n" % e.message)
+
+    log.info("Switched %i Loaders of the %i\n" % (len(representations),
+                                                  len(containers)))
+    # Updating frame range
+    log.debug("\nUpdating frame range ..")
```

Comment:
If Fusion requires the `\n` then this one is ordered differently then the other logs above ;) The new line should then be at the end?

Maybe we can find a way to fix it in the Fusion logger, so it doesn't need the new line explicitly here. (Magic!)